assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentServiceImpl.java

```diff
@@ -286,6 +286,49 @@ public boolean deleteTraineeAssessment(Long assessmentId, Long traineeId) {
     return false;
   }
 
+  private AssessmentDTO updateAssessmentWithNestedDtos(AssessmentDTO assessmentDto) {
+    Assessment assessment = assessmentMapper.toEntity(assessmentDto);
+    AssessmentDetailDTO assessmentDetailDto = null;
+    AssessmentOutcomeDTO assessmentOutcomeDto = null;
+    RevalidationDTO revalidationDto = null;
+
+    if (assessmentDto.getDetail() != null) {
+      if (assessmentDto.getDetail().getId() != null) {
+        assessmentDetailDto = assessmentDetailService.save(assessment,
+            assessmentDto.getDetail());
+      } else {
+        assessmentDetailDto = assessmentDetailService.create(assessment,
+            assessmentDto.getDetail());
+      }
```

Comment:
You could use a ternary if the change in method length is why you've refactored this.

Reply:
Forgot the snippet:
```java
assessmentDetailDto = assessmentDto.getDetail().getId() != null ?
          assessmentDetailService.save(assessment, assessmentDto.getDetail())
          : assessmentDetailService.create(assessment, assessmentDto.getDetail());
```

Reply:
@JosephKelly , there was a code smell complaining about the complexity so I refactored that bit of codes into a new method. Ternary looks much neater, thanks!